arangod/Utils/Cursor.h

```diff
@@ -145,10 +163,13 @@ class Cursor {
  protected:
   CursorId const _id;
   size_t const _batchSize;
+  size_t _currentBatchId;
   double _ttl;
   std::atomic<double> _expires;
   bool const _hasCount;
   bool _isDeleted;
+  bool _isRetriable;
   std::atomic<bool> _isUsed;
+  std::pair<size_t, VPackBuilder> _currentBatchResult;
```

Comment:
I don't like storing a full `Builder` here. All we need is the slice data contained in the Builder, which the Builder stores in its underlying Buffer.
So why don't we just store a Buffer here?